docs/packages/xstate-solid/index.md

```diff
@@ -0,0 +1,382 @@
+              <button onClick={() => send('FETCH', { query: 'something' })}>
+                Search for something
+              </button>
+            </Match>
+            <Match when={state.matches('loading')}>
+              <div>Searching...</div>
+            </Match>
+            <Match when={state.matches('success')}>
+              <div>Success! Data: {state.context.data}</div>
+            </Match>
+            <Match when={state.matches('failure')}>
+              <div>
+                <p>{state.context.error.message}</p>
+                <button onClick={() => send('RETRY')}>Retry</button>
+              </div>
+            </Match>
+          </Switch>
+  );
+};
+```
+
+## Matching States
+
+When using [hierarchical](https://xstate.js.org/docs/guides/hierarchical.html) and [parallel](https://xstate.js.org/docs/guides/parallel.html) machines, the state values will be objects, not strings. In this case, it is best to use [`state.matches(...)`](https://xstate.js.org/docs/guides/states.html#state-methods-and-getters).
+
+The SolidJS [Switch and Match Components]() are ideal for this use case:
+
+```jsx
+const Loader = () => {
+  const [state, send] = useMachine(/* ... */);
+
+  return (
+          <div>
+            <Switch fallback={null}>
+              <Match when={state.matches('idle')}>
+                <Loader.Idle />
+              </Match>
+              <Match when={state.matches({ loading: 'user' })}>
+                <Loader.LoadingUser />
+              </Match>
+              <Match when={state.matches({ loading: 'friends' })}>
+                <Loader.LoadingFriends />
+              </Match>
+            </Switch>
+          </div>
+  );
+};
+```
+
+## Persisted and Rehydrated State
+
+You can persist and rehydrate state with `useMachine(...)` via `options.state`:
+
+```js
+// ...
+
+// Get the persisted state config object from somewhere, e.g. localStorage
+const persistedState = JSON.parse(localStorage.getItem('some-persisted-state-key')) || someMachine.initialState;
+
+const App = () => {
+  const [state, send] = useMachine(someMachine, {
+    state: persistedState // provide persisted state config object here
+  });
+
+  // state will initially be that persisted state, not the machine's initialState
+
+  return (/* ... */)
+}
+```
+
+## Services
+
+The `service` created in `useMachine(machine)` can be referenced as the third returned value:
+
+```js
+//                  vvvvvvv
+const [state, send, service] = useMachine(someMachine);
+```
+
+You can subscribe to that service's state changes with the [`createEffect` hook](https://www.solidjs.com/docs/latest/api#createeffect):
+
+```js
+// ...
+
+createEffect(() => {
+  const subscription = service.subscribe((state) => {
+    // simple state logging
+    console.log(state);
+  });
+
+  onCleanup(() => subscription.unsubscribe());
+}); // note: service should never change
+```
+
+Or by using the [`from` utility](https://www.solidjs.com/docs/latest/api#from)
```

Comment:
Could we add a note about its drawback? The drawback being that this isn't deeply reactive